src/main.ts

```diff
@@ -41,14 +42,17 @@ async function run(): Promise<void> {
     core.debug(gitOutput);
     core.debug('🐱🐱🐱🐱🐱 ^^^ gitOutput');
 
-    if ((!gitOutput) || gitError) {
+    if ((failOnEmpty && !gitOutput) || gitError) {
       if (!gitOutput)
         {core.setFailed('git stdout: ∅');}
       if (gitError)
         {core.setFailed(`git stderr: ${gitError}`);}
       return;
     }
 
+    if (!gitOutput)
```

Comment:
Optional: I not a huge fan of figuring out the code paths here, does this read a little easier:

```
if (gitError) {
	core.setFailed(`git stderr: ${gitError}`)
	return;
}

if (!gitOutput) {
	if (failOnEmpty) {
		core.setFailed('git stdout: ∅');
	}
	return;
}
```

Reply:
I was trying to cover the case where both conditions are true, though admittedly I haven't actually tested calling `setFailed` more than once. I added some comments in d9f360b094c147027eb47031fb96b9d988560795. S'okay?